Editing/input-event-dataTransfer-explainer.md

```diff
@@ -0,0 +1,158 @@
+- Introduce new input types or event interfaces.
+
+## Motivation
+The W3C Input Events Level 2 specification requires that [`dataTransfer`](https://html.spec.whatwg.org/multipage/dnd.html#datatransfer) be available for certain input types. Prior to this feature, Chromium did not expose [`dataTransfer`](https://html.spec.whatwg.org/multipage/dnd.html#datatransfer) on [`InputEvent`](https://w3c.github.io/input-events/#interface-InputEvent) for [`insertFromPaste`, `insertReplacementText`, and `insertFromDrop`](https://w3c.github.io/input-events/#overview), resulting in inconsistent behavior across browsers and limiting developer capabilities.
+
+This feature addresses:
+- Spec compliance.
+- Interoperability with Safari and Firefox.
+- Developer needs for accessing drag-and-drop and clipboard data in editable content.
+
+### Code Example
+#### Before the Fix
+```javascript
+editor.addEventListener("input", (event) => {
+  console.log(event.inputType); // e.g., "insertFromDrop"
+  console.log(event.dataTransfer); // null
+});
+```
+
+#### After the Fix
+```javascript
+editor.addEventListener("input", (event) => {
+  console.log(event.inputType); // e.g., "insertFromDrop"
+  console.log(event.dataTransfer); // non-null
+  if (event.dataTransfer) {
+    console.log("Dropped plain data:", event.dataTransfer.getData("text/plain"));
+    console.log("Dropped html data:", event.dataTransfer.getData("text/html"));
+  }
+});
+```
+In the implementation, [`dataTransfer`](https://html.spec.whatwg.org/multipage/dnd.html#datatransfer) property is attached only when the target is a [`contenteditable`](https://html.spec.whatwg.org/multipage/interaction.html#attr-contenteditable) element, excluding form controls ([`input`](https://html.spec.whatwg.org/multipage/input.html#the-input-element) or [`textarea`](https://html.spec.whatwg.org/multipage/form-elements.html#the-textarea-element)).
+
+## Considered Alternatives
+- Keeping the current behavior: This would continue to violate the spec and limit developer capabilities (currently developers can get [`dataTransfer`](https://html.spec.whatwg.org/multipage/dnd.html#datatransfer) only if they listen to [`beforeinput`](https://w3c.github.io/input-events/#event-type-beforeinput) event).
+- Exposing [`dataTransfer`](https://html.spec.whatwg.org/multipage/dnd.html#datatransfer) for all elements: This was rejected to preserve existing behavior for form controls and avoid unintended side effects.
+
+## Security and Privacy
+This proposal has no known impact on accessibility or privacy and does not alter the permission or security model. The [`dataTransfer`](https://html.spec.whatwg.org/multipage/dnd.html#datatransfer) property is exposed only when the browser determines it is appropriate based on the input event type and context. It is scoped to [`contenteditable`](https://html.spec.whatwg.org/multipage/interaction.html#attr-contenteditable) elements and does not alter drag-and-drop security model.
```

Comment:
You could also say here that no new information is exposed to sites because the same information is also exposed during the `beforeinput` event.